Installer/update-vs.ps1

```diff
@@ -0,0 +1,99 @@
+    write-host -ForegroundColor Red 'fatal: Access Denied'
+    write-host -ForegroundColor Yellow 'Administrator privilages are required, please run again with administrator privilages.'
+    
+    exit -2
+  }
+}
+
+$current_directory = get-location
+$script_name = $MyInvocation.MyCommand.Name
+$installer_path = "$current_directory\$installer_name"
+$script_path = "$current_directory\$script_name"
+$vs_installer_path = "${Env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer"
+
+write-host -ForegroundColor Cyan $title
+
+if (-not (test-path($installer_path))) {
+  write-host "fatal: unable to find $installer_name."
+  
+  write-host -NoNewLine 'Press any key to continue...';
+  $null = $Host.UI.RawUI.ReadKey('NoEcho,IncludeKeyDown');
+  
+  exit -1
+}
+
+if (-not (test-path($vs_installer_path))) {
+  write-host -ForegroundColor Yellow 'Visual Studio was not detected on this machine.'
+  
+  write-host -NoNewLine 'Press any key to continue...';
+  $null = $Host.UI.RawUI.ReadKey('NoEcho,IncludeKeyDown');
+  
+  exit 1
+}
+
+# Ensure that this script is executing elevated
+check_access
+
+write-host 'Scanning for instllations of Visual Studio to update'
+write-host 
+
+$vswhere_path = "$vs_installer_path\vswhere.exe"
+$vswhere_cmd = "& '$vswhere_path' -all -prerelease -nologo -utf8 -version '[15.0, 16.0)'"
+
+# Run vswhere and parse its output.
+$vswhere_output = (iex "$vswhere_cmd | findstr /I 'installation'") | out-string
+$reader = new-object -TypeName System.IO.StringReader -ArgumentList $vswhere_output
+[int]$success_count = 0
+[int]$failure_count = 0
+$installation_name
+$installation_path
+
+while ($line = $reader.ReadLine()) {
+  if ($line -like 'installationName: *') {
+    $installation_name = $line.SubString('installationName: '.Length)
+  } elseif ($line -like 'installationPath: *') {
+    $installation_path = $line.SubString('installationPath: '.Length)
+    
+    write-host -ForegroundColor Cyan "Found $installation_name ($installation_path)"
+    
+    $installation_path  = "$installation_path\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer\Git"
+    
+    # Execute the command line installer script.
+    $installer_cmd = "& '$installer_path' install --passive --path '$installation_path'"
+    iex "$installer_cmd"
+    
+    if ($lastExitCode -ne 0) {
+      $failure_count = $failure_count + 1
+      write-host -ForegroundColor Yellow "Failed to update '$installation_path'"
+    } else {
+      $success_count = $success_count + 1
+    }
+    
+    write-host 
+  }
+}
+
+[int]$total_count = $failure_count + $success_count
+
+if ($total_count -gt 0) {
+  write-host -ForegroundColor Cyan -NoNewLine "$total_count Visual Studio installation(s) found"
+  if ($success_count -gt 0) {
+    write-host -ForegroundColor Cyan -NoNewLine ", $success_count successfully updated"
+  }
+  if ($failure_count -gt 0) {
+    write-host -ForegroundColor Cyan -ForegroundColor Yellow -NoNewLine ", $failure_count failed to updated"
```

Comment:
write-warning?

Reply:
`write-warning` did not even know it existed... learn something new every day 🙂

Reply:
hmm... the `Write-Warning` commandlet doesn't allow for colorization, meaning that the output could be visually lost. Personally, I prefer the colorization for ease of consumption. Is there a legit reason for all of these `Write-` types (I assume that there must be)?

Command | Description
----------- | -------------
`Write-Debug` | Writes a debug message to the console.
`Write-Error` | Writes an object to the error stream.
`Write-Host` | Writes customized output to a host.
`Write-Information` | Specifies how Windows PowerShell handles information stream data for a command.
`Write-Output` | Sends the specified objects to the next command in the pipeline.
`Write-Progress` | Displays a progress bar within a Windows PowerShell command window.
`Write-Verbose` | Writes text to the verbose message stream.
`Write-Warning` | Writes a warning message.